src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -841,6 +844,30 @@ ImmutableHashSet<string> ComputeSupportedLiveDiagnosticIds()
                 }
             }
 
+            internal string? GetHelpLinkUri(ProjectId projectId, string diagnosticId)
+            {
+                var infoCache = _owner._diagnosticService.AnalyzerInfoCache;
+                var project = Solution.GetProject(projectId);
+                if (project is null)
+                {
+                    return null;
+                }
+
+                foreach (var analyzersPerReference in project.Solution.State.Analyzers.CreateDiagnosticAnalyzersPerReference(project))
+                {
+                    foreach (var analyzer in analyzersPerReference.Value)
+                    {
```

Comment:
I'm a little bit concerned about performance here. But I see similar nested loops in this class, so I'm assuming this is okay.